### DISCUSSION TITLE
Modify export query using options

### DISCUSSION THREAD

**Discussion by smortexa** [filamentphp/filament#12519] [Ideas]
The `modifyQuery` static method currently lacks the capability to access options, which prevents the filtering of the export query.

**Comment by scottzirkel** [comment]
Yeah, I'm struggling with this as well.

**Comment by lukeholder** [comment]
Has anyone found a workaround for this??

**Reply by lukeholder** [reply]
I found that I can only modify the query from the context of the export action with:

`modifyQueryUsing`

```php
 Actions\ExportAction::make()
  ->label('Legacy Export Price List')
  ->exporter(LegacyPriceListItemExporter::class)
  ->options([
    'price_list_id' => $this->record->getKey(), // cant use this option in static export ::modifyQuery()
  ])
  ->modifyQueryUsing(fn (Builder $query) => $query->where('price_list_id', $this->record->getKey())) // need to pass it in from here
  ->columnMapping(false),
```

**Reply by DavidKoLab** [reply]
I'm not sure if this is what you are asking, but jus for your reference. Exporter method getOptionsFormComponents results can be injected as $data in modifyQueryUsing method
```
ExportAction::make()
    ->exporter(ProductExporter::class)
    ->modifyQueryUsing(fn (Builder $query, array $data) => $query->where('is_active', $data['is_active']))
```

[doc_here](https://filamentphp.com/docs/3.x/actions/advanced)
[getOptionsFormComponents_doc](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/export#using-export-options)

**Reply by lukeholder** [reply]
@DavidKoLab right, but you cant access them in the static`LegacyPriceListItemExporter::modifyQuery()` even though the docs list it as an alternative to modifying it in the action: https://filamentphp.com/docs/3.x/actions/prebuilt-actions/export#modifying-the-export-query

**Comment by curleytd** [comment]
Anybody ever find a solution for this? Ended up going the route of making a custom action to support the options being included in the query but would love a more direct approach with support.
